### PR TITLE
Refactoring REST API

### PR DESCRIPTION
**Pull request by anuruddhal**
Following modifications were done.
  
>Catch specific/custom exception - customize the log messages according to exception type (addCartridge(), updateCartridge() etc.)
    >getAvailableCartridges() - remove obsolete code & simplify
    >Remove obsolete code related to billing
    >getCartridgesByFilter() remove hard-coded strings
    >Change method parameters to primitive boolean
    >Refactor isAlreadySubscribed() method
    >Add missing method comments
    >Refactor validateDeploymentPolicy(), getApplicationRuntime(), getApplicationSignUps()
    >Make private static log to be final
    >Rename StratosUserManager as a util class and make the methods static
    >Removed ErrorResponseBean and SucessResponseBean and add StatusResponseBean Class.

### CODE REVIEW

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java [hunk 1]
```diff
@@ -123,13 +128,31 @@ public static void addCartridge(CartridgeBean cartridgeDefinition) throws RestAP
                         "Successfully added cartridge: [cartridge-type] %s ",
                         cartridgeDefinition.getType()));
             }
-        } catch (Exception e) {
+        }catch (CloudControllerServiceCartridgeAlreadyExistsExceptionException e) {
```
Comment (prabathabey) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Fix formatting issues.

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java [hunk 2]
```diff
@@ -123,13 +128,31 @@ public static void addCartridge(CartridgeBean cartridgeDefinition) throws RestAP
...
+            String msg = "Could not add cartridge";
+            log.error(msg, e);
+            throw new RestAPIException(msg);
+        } catch (CloudControllerServiceInvalidCartridgeDefinitionExceptionException e) {
```
Comment (prabathabey) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: If there's nothing specific to be logged when each of these exceptions was caught, might be a good option to use "|" notation to have them all clustered into one catch block, which would improve readability of the code.

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java [hunk 3]
```diff
@@ -146,16 +169,31 @@ public static void updateCartridge(CartridgeBean cartridgeDefinition) throws Res
...
             }
         } catch (CloudControllerServiceCartridgeDefinitionNotExistsExceptionException e) {
-            String msg = "No cartridge definition exists with this definition.Please use the POST method to add the cartridge";
+            String msg = "Could not add cartridge";
             log.error(msg, e);
             throw new RestAPIException(msg);
-        } catch (Exception e) {
-            String msg = "Could not update cartridge " + e.getLocalizedMessage();
+        } catch (CloudControllerServiceInvalidCartridgeDefinitionExceptionException e) {
+            String msg = "Could not add cartridge";
```
Comment (prabathabey) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Same goes here as well.

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java (hunk 2 above)
Comment (anuruddhal) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Fixed with PR https://github.com/apache/stratos/pull/405

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java (hunk 3 above)
Comment (anuruddhal) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Fixed with https://github.com/apache/stratos/pull/405

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java (hunk 1 above)
Comment (anuruddhal) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Already fixed in the latest code.

**Pull request merged**